src/expr/impl/src/scalar/vnode.rs

```diff
@@ -23,17 +23,16 @@ use risingwave_expr::{build_function, Result};
 
 #[derive(Debug)]
 struct VnodeExpression {
-    dist_key_indices: Vec<usize>,
+    children: Vec<BoxedExpression>,
+    all_indices: Vec<usize>,
 }
 
 #[build_function("vnode(...) -> int2")]
 fn build(_: DataType, children: Vec<BoxedExpression>) -> Result<BoxedExpression> {
-    let dist_key_indices = children
-        .into_iter()
-        .map(|child| child.input_ref_index().unwrap())
-        .collect();
-
-    Ok(Box::new(VnodeExpression { dist_key_indices }))
+    Ok(Box::new(VnodeExpression {
+        all_indices: (0..children.len()).collect(),
```

Comment:
I am confused by this field, although it's renamed from `dist_key_indices`. Can you please exemplify and explain its usage?

Reply:
It's a parameter required by `VirtualNode::compute_row` to project the distribution key columns. But it's true that this can be done in the callsite by simply attaching `.project`. Will change the signature.

Reply:
Oh, it's also used by `VirtualNode::compute_chunk`.

Reply:
`input.project(dist_key_indices)` is changed to `children.eval(input)` so no `dist_key_indices` needed anymore.

`all_indices` is a new field introduced to reduce redundant vec allocation.

Reply:
Maybe it's better to rename `children` to `dist_key_exprs` or sth.